src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -0,0 +1,54 @@
+package wooteco.subway.service;
+
+import java.util.List;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineDto;
+import wooteco.subway.dto.LineRequest;
+
+@Service
+public class LineService {
+
+    private final LineDao lineDao;
+
+    public LineService(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    public Line save(LineRequest lineRequest) {
+        validDuplicatedName(lineRequest.getName());
+        Line line = new Line(lineRequest.getName(), lineRequest.getColor());
+        Long id = lineDao.save(line);
+        return new Line(id, line.getName(), line.getColor());
+    }
+
+    public void update(Long id, LineRequest lineRequest) {
+        validDuplicatedName(lineRequest.getName());
+        lineDao.update(id, lineRequest);
+    }
+
+    private void validDuplicatedName(String name) {
+        if (lineDao.countByName(name) > 0) {
+            throw new IllegalArgumentException("중복된 이름이 존재합니다.");
+        }
+    }
```

Comment:
```suggestion
    private void validDuplicatedName(String name) {
        if (lineDao.existsByName(name)) {
            throw new IllegalArgumentException("중복된 이름이 존재합니다.");
        }
    }
```
이렇게 하면 매직넘버를 줄일 수 있을 것 같아요